webhook/resourcesemantics/validation/validation_admit.go

```diff
@@ -38,9 +38,25 @@ var errMissingNewObject = errors.New("the new object may not be nil")
 
 // Callback is a generic function to be called by a consumer of validation
 type Callback struct {
-	Func func(ctx context.Context, unstructured *unstructured.Unstructured) error
+	// f is the function to be called
+	f func(ctx context.Context, unstructured *unstructured.Unstructured) error
 
-	SupportedVerbs map[webhook.Operation]bool
+	// supported are the verbs supported for the callback.
+	// The function will only be called on these acitons.
+	supported map[webhook.Operation]bool
+}
+
+// NewCallback creats a new callback function to be invoked on supported vebs.
+func NewCallback(function func(context.Context, *unstructured.Unstructured) error, supportedVerbs []webhook.Operation) (c *Callback) {
+	m := make(map[webhook.Operation]bool)
```

Comment:
```suggestion
	m := make(map[webhook.Operation]struct{})
```
Doesn't really matter here but in general that's how go does sets.

Reply:
sure. Same thing to me - I think I saw both bool and struct referenced on golang forums